//https://bugs.eclipse.org/bugs/show_bug.cgi?id=124853 - variation
public void test035() {
	this.runConformTest(
			new String[] {
				"X.java",
				"public class X {\n" + 
				"	public static void main(String[] args) {\n" + 
				"		try {\n" + 
				"			new X().bar();\n" + 
				"		} catch(Exception e){\n" + 
				"			System.out.println(\"[end]\");\n" + 
				"		}\n" + 
				"	}\n" + 
				"	Object bar() {\n" + 
				"		try {\n" + 
				"			System.out.print(\"[try]\");\n" + 
				"			return this;\n" + 
				"		} catch(Exception e){\n" + 
				"			System.out.print(\"[WRONG CATCH]\");\n" + 
				"		} finally {\n" + 
				"			System.out.print(\"[finally]\");\n" + 
				"			foo();\n" + 
				"		}\n" + 
				"		return this;\n" + 
				"	}\n" + 
				"	Object foo() {\n" + 
				"		throw new RuntimeException();\n" + 
				"	}\n" + 
				"}\n",
			},
			"[try][finally][end]");
}

